lib/internal/bootstrap_node.js

```diff
@@ -411,8 +411,38 @@
 
   // Load preload modules
   function preloadModules() {
-    if (process._preload_modules) {
-      NativeModule.require('module')._preloadModules(process._preload_modules);
+    var allPreloads = process._preload_modules;
+    var isSetUidRoot = false;
+
+    if (process.platform !== 'win32') {
+      isSetUidRoot = process.getuid() !== process.geteuid() ||
```

Comment:
Look for how OPENSSL_CONF is loaded from env, in such a way that its ignored if the IDs aren't safe, much as you are manually trying to check here.